cmd/snap/cmd_run.go

```diff
@@ -175,9 +177,17 @@ func runSnapConfine(info *snap.Info, securityTag, snapApp, command, hook string,
 		logger.Noticef("WARNING: cannot create user data directory: %s", err)
 	}
 
-	cmd := []string{
-		filepath.Join(dirs.LibExecDir, "snap-confine"),
+	// use snap-confine from core if it is available
```

Comment:
We also need to use the dynamic linker from core as snap-confine dynamically links to libraries that are not available on the host (e.g. xfs, apparmor).